src/instructions/mod.rs

```diff
@@ -10,9 +10,34 @@ pub mod tables;
 pub mod tlb;
 
 /// Halts the CPU until the next interrupt arrives.
-#[inline(always)]
+#[inline]
 pub fn hlt() {
     unsafe {
         asm!("hlt" :::: "volatile");
     }
 }
+
+/// Emits a 'magic breakpoint' instruction for the [Bochs](http://bochs.sourceforge.net/) CPU
+/// emulator. Make sure to set `magic_break: enabled=1` in your `.bochsrc` file.
+#[inline]
+pub fn bochs_breakpoint() {
+    unsafe {
+        asm!("xchgw %bx, %bx" :::: "volatile");
+    }
+}
+
+/// Gets the current instruction pointer. Note that this is only approximate as it requires a few
+/// instructions to execute.
+#[inline]
+pub fn read_rip() -> u64 {
```

Comment:
I think this function fits better in the `registers` module, given that `rip` is a CPU register.